src/app/wallet/wallet.service.ts

```diff
@@ -34,4 +35,14 @@ export class WalletService {
             'pair', pairingCode
         ]), API.tokenHeaders());
     }
+
+    getTransactionHistory() {
+        const transactionResponse = this.http.get<TransactionList>(
+            API.generateRoute(this.transactionHistoryEndpoint),
+            API.tokenHeaders());
+        transactionResponse.subscribe((res: TransactionList) => {
+            UserStatusStorage.transactionData = res;
+        });
+        return transactionResponse;
+    }
```

Comment:
As we spoke earlier, avoid subscribing to subscription inside service as this results with multiple HTTP calls. Variable `transactionResponse` is not a response, but [cold subscription](https://codingthesmartway.com/getting-started-with-rxjs-part-3-hot-and-cold-observables/).

```suggestion
    getTransactionHistory() {
        return this.http.get<TransactionList>(
            API.generateRoute(this.transactionHistoryEndpoint), API.tokenHeaders())
            .pipe(tap((res: TransactionList) => UserStatusStorage.transactionData = res)));
    }
```

Check the code I suggested if it's correct.